test/BlockManager.js

```diff
@@ -899,5 +899,155 @@ describe('BlockManager', function () {
 
       await assertRevert(tx, 'Proposed Alternate block is identical to proposed block');
     });
+    it('Blocks should be proposed according to iteration', async function () {
```

Comment:
also write tests about https://github.com/razor-network/contracts/pull/429/files#diff-ef39b70abd99dba15254b650eeab6fe94780b7804fbd0715169e10e8ad67e198R243

Reply:
Wrote a test case on this